server/storage/mvcc/kvstore_test.go

```diff
@@ -620,10 +620,11 @@ func TestHashKVWhenCompacting(t *testing.T) {
 
 	select {
 	case <-donec:
-		wg.Wait()
 	case <-time.After(10 * time.Second):
 		testutil.FatalStack(t, "timeout")
```

Comment:
If it fatal, L627 is still working?

Reply:
Good catch. Updated.

The key point is to ensure all goroutines to finish before executing the deferred function.